packages/core/admin/ee/server/controllers/workflows/index.js

```diff
@@ -0,0 +1,42 @@
+'use strict';
+
+const { merge } = require('lodash/fp');
+const { getService, mapObject } = require('../../utils');
+
+function sanitizeWorkflowQuery(query = {}) {
+  return mapObject(query, {
+    pick: ['workflow_id'],
+    rename: { workflow_id: 'id' },
+  });
+}
+
+module.exports = {
+  /**
+   * List all workflows
+   * @param {import('koa').BaseContext} ctx - koa context
+   */
+  async find(ctx) {
+    const query = sanitizeWorkflowQuery(merge(ctx.query, ctx.params));
```

Comment:
Yup is for validation, here we just have a function to sort and filter the input parameters and only get the ones that are relevant.

Reply:
Params are the parameters in the url, and query in the querystring. Why not merge them as they act as the same input?

Reply:
I don't understand why extracting and merging them, they are already easily accessible.
If the id of the entity is needed it's params.id and that's it.
When there is an input to validate/sanitize we just validate it against a yup schema without transforming the data.
An example in the code that I can be followed:
```js
  async update(ctx) {
    const { id } = ctx.params;
    const { body } = ctx.request;

    const roleService = getService('role');

    await validateRoleUpdateInput(body);

    const role = await roleService.findOne({ id });

    if (!role) {
      return ctx.notFound('role.notFound');
    }

    if (role.code === SUPER_ADMIN_CODE) {
      throw new ApplicationError("Super admin can't be edited.");
    }

    const updatedRole = await roleService.update({ id }, body);
    const sanitizedRole = roleService.sanitizeRole(updatedRole);

    ctx.body = {
      data: sanitizedRole,
    };
  },
  ```
I find it simpler.

`Params` and `query` are different kind of inputs, I don't think they should be merged as if they were the same. With the logic of input we would also merge `body`. I think it's cleaner to let them separated. Collisions can happen, even if it can be avoided it doesn't feel clean to mix.

Reply:
The sanitizeQuery function is there to standardise all the inputs received from the various controllers. It allows us to extract and retrieve only the query parameters that are useful to us. `body` isn't a parameter (can be, but shouldn't be), it's data that is sent to the API.
We could write `const { id } = ctx.params;` in each controller, but if, for some reason, id were to change its name (or any other parameter), we would have to refactor all controllers functions.
I understand that for the moment, given the possibilities offered by find and findOne, it does not seem very useful. But I prefer to use functions as much as possible to avoid later rework.

Also, as I try to apply the principle of never trusting incoming data, it seems appropriate to filter out unwanted data from the beginning of the controller execution.
I actually think we can add a yup validation on top of that to validate the content of the parameters. 👍

Reply:
Not only for this part but in general, my opinion is we should follow what the rest of Strapi is doing. And if we want to improve how controllers pick and validat query params, we should have a discussion in the backend sync with more people.

We should have a common way of doing this in Strapi, so understanding our code will be simpler.

So, summarizing:
- Could we use a format that's common in strapi for now? Using yup would folow the principle of not trusting incoming data & would give better errors for the user.
- If in some cases we want to use other format like this one, could we bring this in a backend sync , or discuss it somewhere?
- Whatever the decision, could we document this format in the Contributor docs? So we have a common place to read a template of how to define a controller in strapi.

wdyt?

Reply:
On top of everything that has been said, the `find` route that references this controller method doesn't have any params. So why are we trying to extract potential values from it?

Let's keep it simple & only implement what we need right now. If at some point we need to reference the params, we can do it quickly later.

What do you think?

Reply:
Sure, I will redo this.